### PR TITLE
Add ``staticcall`` to ``address``.

### LINKED ISSUE #4768: Add `address.staticcall(...)`

### PR DESCRIPTION
**Pull request by ekpyron**
Fixes #4768.

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -4215,6 +4228,53 @@ BOOST_AUTO_TEST_CASE(generic_delegatecall)
...
+					function pureFunction(uint256 p) public pure returns (uint256) { return p; }
+					function viewFunction(uint256 p) public view returns (uint256) { return p + x; }
+					function nonpayableFunction(uint256 p) public returns (uint256) { x = p; return x; }
+					function assertFunction(uint256 p) public view returns (uint256) { assert(x == p); return x; }
+				}
+				contract C {
+					function f(address a) public view returns (bool)
+					{
+						bytes4 signature = bytes4(bytes32(keccak256("pureFunction(uint256)")));
+						return a.staticcall(abi.encodeWithSelector(signature, 23));
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp]: Why not use `encodeWithSignature`?

Comment (ekpyron) [reply test/libsolidity/SolidityEndToEndTest.cpp]: No reason - just copied the test from ``delegatecall`` - I'll change it.

**Review by chriseth — DISMISSED** [review]
Please add a syntax check that tests if `address(1).staticcall.value` exists or not.

File: test/libsolidity/SolidityEndToEndTest.cpp (hunk 1 above)
Comment (ekpyron) [reply test/libsolidity/SolidityEndToEndTest.cpp]: Done.

Comment (ekpyron) [reply test/libsolidity/SolidityEndToEndTest.cpp]: I left it the same as before for ``delegatecall``, though.

File: libsolidity/codegen/ExpressionCompiler.cpp [hunk 2]
```diff
@@ -1172,6 +1173,7 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 				case FunctionType::Kind::BareCall:
 				case FunctionType::Kind::BareCallCode:
 				case FunctionType::Kind::BareDelegateCall:
+				case FunctionType::Kind::BareStaticCall:
```
Comment (chriseth) [inline libsolidity/codegen/ExpressionCompiler.cpp:1176]: I don't think these should be here at all - will open a separate PR.

**Review by chriseth — APPROVED** [review]

**Review by chriseth — CHANGES_REQUESTED** [review]
This needs more documentation changes, at least by grepping `callcode` I can find more ocurrences.

**Comment by ekpyron** [comment]
Added some more documentation.

**Review by chriseth — APPROVED** [review]

**Pull request merged**